### DISCUSSION TITLE
NodejsFunction support?

### DISCUSSION THREAD

**Discussion by sumort** [mrgrain/cdk-esbuild#667] [Q&A] [answered]
Hi, I just wanted to confirm if this can be used with the [NodejsFunction](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs.NodejsFunction.html) construct? 

I saw some examples for the [Function](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html)  construct but as I was comparing the 2, I noticed that for `Function`, it accepts `code` for the source code in comparison to `NodejsFunction` which has `entry` which is a string to indicate the path to the code. So I'm not sure if there is a way to use this with the `NodejsFunction` construct. 

Thank you in advance!

**Comment by mrgrain** [comment, marked as answer]
This package _replaces_ usage of `NodejsFunction`. Instead you create a `Function` and provide `TypeScriptCode` to the `code` property.
See this example: https://github.com/mrgrain/cdk-esbuild#aws-lambda-serverless-function


Both `NodejsFunction` and `@mrgrain/cdk-esbuild` solve a similar problem ("bundling TS code"), but in a different way.

**Comment by sumort** [comment]
Copy that, thank you! @mrgrain
